### PR TITLE
Collision System None & Motion Planning

### PR DESCRIPTION
**Pull request by fundies**
fix for https://github.com/enigma-dev/enigma-dev/issues/943

### CODE REVIEW

**Comment by fundies** [comment]
fixes https://github.com/enigma-dev/enigma-dev/issues/456 also

File: ENIGMAsystem/SHELL/Universal_System/Extensions/MotionPlanning/motion_planning_struct.h [hunk 1]
```diff
@@ -35,6 +35,14 @@ using std::multimap;
 #  error This file includes non-ENIGMA STL headers and should not be included from SHELLmain.
 #endif
 
+#ifdef COLLISION_NONE
```
Comment (JoshDreamland) [inline ENIGMAsystem/SHELL/Universal_System/Extensions/MotionPlanning/motion_planning_struct.h]: [Suggested fix](https://github.com/fundies/enigma-dev/compare/master...JoshDreamland:patch-2).

File: CommandLine/programs/emake/link.cpp [hunk 2]
```diff
@@ -12,7 +12,8 @@
 * Foundation, version 3 of the License, or (at your option) any later version.
 *
 * ENIGMA is distributed in the hope that it will be useful, but WITHOUT ANY
-* WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+* WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+*A
```
Comment (JoshDreamland) [inline CommandLine/programs/emake/link.cpp]: This is one of the reasons I'm fearful of just running Astyle. Also the enormous amount of space it adds after newlines in parentheses.

**Comment by JoshDreamland** [comment]
In my opinion, the Astyle/clang-format just made everything worse. It made some things better, but it fucked up some preprocessor indentation, added heaps of whitespace to comments it wrapped and to line breaks made in parentheticals, as well as just uglifying the main license comment, because it's smart enough to wrap a comment further, but not smart enough to unwrap existing breaks.

**Pull request closed without merging**